gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/InsertAllRequestTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+public class InsertAllRequestTest {
+
+  private static final Map<String, Object> CONTENT1 =
+      ImmutableMap.<String, Object>of("key", "val1");
+  private static final Map<String, Object> CONTENT2 =
+      ImmutableMap.<String, Object>of("key", "val2");
+  private static final List<InsertAllRequest.RowToInsert> ROWS =
+      ImmutableList.of(new InsertAllRequest.RowToInsert(CONTENT1),
+          new InsertAllRequest.RowToInsert(CONTENT2));
+  private static final List<InsertAllRequest.RowToInsert> ROWS_WITH_ID =
+      ImmutableList.of(new InsertAllRequest.RowToInsert("id1", CONTENT1),
+          new InsertAllRequest.RowToInsert("id2", CONTENT2));
+  private static final TableId TABLE_ID = TableId.of("dataset", "table");
+  private static final boolean SKIP_INVALID_ROWS = true;
```

Comment:
Nit: for the Boolean values, I think it would be nice to vary the values.  Say someone in the future accidentally always sets SKIP_INVALID_ROWS to true or IGNORE_UNKNOWN_VALUES to false, then this test wouldn't catch that.